test/shared/src/main/scala/zio/test/mock/MockableRandom.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.mock
+
+import zio.{ Chunk, UIO }
+import zio.random.Random
+
+object MockableRandom {
+
+  object Service {
+    object NextBoolean       extends Method[Nothing, Boolean]
+    object NextBytes         extends Method[Int, Chunk[Byte]]
+    object NextDouble        extends Method[Nothing, Double]
+    object NextFloat         extends Method[Nothing, Float]
+    object NextGaussian      extends Method[Nothing, Double]
+    object NextInt           extends Method[Int, Int]
+    object NextInt1          extends Method[Nothing, Int]
```

Comment:
Hmm.. I'm wondering how to deal with method overloading. For the time being I just added a numeric suffix, but it's kinda arbitrary. Here the intuitive thing would be to use `NextInt` or `NextInt0` for the one without input and `NextInt1` for the one taking one input.... BUT:

```
def add(a: Long, b: Long): Long
def add(a: Int, b: Long): Long
def add(a: Int, b: Int): Long
```

are also valid method overloads, in which case the number of arguments does not help to uniquely identify the method.

It's not a problem for manualy created tags like here, but I'm wondering what strategy we should take with `Mockable` macro?

1) like above, simply add numeric suffix, even though the number has completely no meaning
* NextInt takes 1 param and NextInt1 takes 0 params
* but then NextLong takes 0 params and NextLong1 takes 1 param)

2) do not automatically handle overloaded methods with macro, add configuration param to the `@Mockable` annotation allowing the user to instruct the macro how to deal with overloads (this should still result in less boilerplate then manually writing the machinery)

@adamgfraser @jdegoes

Reply:
@ioleo 

I'd just use numbers, probably. Yes, they are arbitrary, but a macro could allow this syntax:

```scala
expect(zio.clock.currentTime, ...)
```

then you can use Scala to choose the method and just look up which index you need for overloads (probabbly easier said than done!).

Reply:
@jdegoes I'll add this as an enchancement proposal to macros project. For now I've decided to go with:
```
object overloadedMethod {
  object _0 extends Method[A, B]
  object _1 extends Method[B, B]
  object _2 extends Method[C, B]
  // ...
}
```

So for each overload we add a `_{0-index}`. This should be nicely discoverable by IDE if you just type `foo.bar.overloadedMethod` followed by dot, you should see a list of possible overloads.

Reply:
Added to `zio-macros` [issue #6](https://github.com/zio/zio-macros/issues/6#issuecomment-536263430).